backend/ctj_api/models.py

```diff
@@ -18,3 +18,68 @@ class Opportunities(models.Model):
 
     class Meta:
         db_table = "opportunities"
+
+
+class CommunityOfPractice(models.Model):
+
+    class PracticeAreas(models.TextChoices):
+        DATA_SCIENCE = 'data_science', 'Data Science'
+        ENGINEERING = 'engineering', 'Engineering'
+        OPS = 'ops', 'Ops'
+        PROJECT_MANAGEMENT = 'project_management', 'Project/Product Management'
+        UI_UX = 'ui_ux', 'UI/UX'
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    practice_area = models.CharField(max_length=50, choices=PracticeAreas.choices)
+    description = models.TextField()
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = "community_of_practice"
+        verbose_name = "Community of Practice"
+        verbose_name_plural = "Communities of Practice"
+
+    def __str__(self):
+        return f"{self.get_practice_area_display()}"
+
+
+class Role(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    title = models.CharField(max_length=50)
+    community_of_practice = models.ForeignKey(CommunityOfPractice, on_delete=models.CASCADE,  related_name="roles")
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = "roles"
+    
+    def __str__(self):
+        return f"{self.title} ({self.community_of_practice.get_practice_area_display()})"
+
+
+class Skill(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    name = models.CharField(max_length=50)
+    roles = models.ManyToManyField('Role', related_name='skills')
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table = "skills"
+
+    def __str__(self):
+        return self.name
+
+
+class Project(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    people_depot_project_id = models.CharField(max_length=255)
```

Comment:
I added this based on the diagram, but I'm unsure what "people depot project id" is.

Reply:
This will point to the `project`'s data from the people_depot's database. CTJ is supposed to get it from there.

CTJ has our own database, and people depot has their own database. In the people depot db, the project will have an id as well. `people_depot_project_id` should point to that id as a string. We will use `people_depot_project_id` to easily find the project in people depot db and get its data. It makes it easier to sync our data together.

It's like if we made a textbook database, we would store the ISBN number of the book to make it easier to google that exact book online later. The ISBN comes from an external system, not ours. `people_depot_project_id` is kind of like the project's ISBN number in people depot database.

This app is tricky because some user data needs to be stored in our db, and some needs to be stored in people_depot. Storing the id of the respective user/project from the people_depot database will hopefully help us sync our data with theirs easier.